drivers/remote/driver.go

```diff
@@ -28,10 +28,25 @@ func newDriver(name string, client *plugins.Client) driverapi.Driver {
 // plugin is activated.
 func Init(dc driverapi.DriverCallback) error {
 	plugins.Handle(driverapi.NetworkPluginEndpointType, func(name string, client *plugins.Client) {
-		c := driverapi.Capability{
-			Scope: driverapi.GlobalScope,
+		// negotiate driver capability with client
+		d := newDriver(name, client)
+		var capResp api.GetCapabilityResponse
+		if err := d.(*driver).call("GetCapability", nil, &capResp); err != nil {
+			log.Errorf("error getting capability for %s due to %v", name, err)
```

Comment:
There have been other breaking changes, so I don't think changing the protocol is inherently a problem.

You could argue that the robustness principle indicates that treating a 404 from the plugin as the default is reasonable. I'm not sure all plugins will return a 404 though -- from memory I don't think it's part of the plugin protocol spec, so there's a chance non-confirming plugins will return an error code, and be left in an unknown state. With that in mind I think it's better to break backwards compatibility and treat all non-conforming responses as errors.